nibabel/funcs.py

```diff
@@ -115,15 +117,19 @@ def concat_images(images, check_affines=True):
     affine = img0.affine
     header = img0.header
     out_shape = (n_imgs, ) + i0shape
```

Comment:
Sorry - I should have explained - it's memory efficient because Python can delete the memory for the individual images as it goes through the loop.  Having said that, in order for that to happen, we need more logic:

```
if is_filename:
    del img
```